src/main/java/com/google/codeu/servlets/OAuth2Callback.java

```diff
@@ -0,0 +1,47 @@
+package com.google.codeu.servlets;
+
+import com.google.api.client.auth.oauth2.AuthorizationCodeFlow;
+import com.google.api.client.auth.oauth2.AuthorizationCodeResponseUrl;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.extensions.appengine.auth.oauth2.AbstractAppEngineAuthorizationCodeCallbackServlet;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class OAuth2Callback extends AbstractAppEngineAuthorizationCodeCallbackServlet {
+
+  private static final long serialVersionUID = 1L;
+
+  @Override
+  protected void onSuccess(HttpServletRequest req, HttpServletResponse resp, Credential credential)
+          throws ServletException, IOException {
+    //log this
+    UserService userService = UserServiceFactory.getUserService();
+
+    String user = userService.getCurrentUser().getEmail();
+    resp.sendRedirect("/user-page.html?user=" + user);
+  }
+
+  @Override
+  protected void onError(HttpServletRequest req, HttpServletResponse resp,
+                         AuthorizationCodeResponseUrl errorResponse)
+          throws ServletException, IOException {
+    String nickname = UserServiceFactory.getUserService().getCurrentUser().getNickname();
+    resp.getWriter().print("<h3>" + nickname + ", why don't you want to play with me?</h1>");
```

Comment:
Lol. Feel free to update this, or maybe discuss with the team what you want to do in this situation.